app/views/layouts/application.html.haml

```diff
@@ -20,6 +20,23 @@
 
     = stylesheet_pack_tag 'application', media: 'all'
     = javascript_pack_tag 'common', integrity: true, crossorigin: 'anonymous'
+
+    = javascript_pack_tag 'features/getting_started', integrity: true, crossorigin: 'anonymous', rel: 'preload'
+
+    = javascript_pack_tag 'features/compose', integrity: true, crossorigin: 'anonymous', rel: 'preload'
+    = javascript_pack_tag 'reducers/compose', integrity: true, crossorigin: 'anonymous', rel: 'preload'
+    = javascript_pack_tag 'reducers/media_attachments', integrity: true, crossorigin: 'anonymous', rel: 'preload'
+    = javascript_pack_tag 'reducers/search', integrity: true, crossorigin: 'anonymous', rel: 'preload'
+
+    = javascript_pack_tag 'features/home_timeline', integrity: true, crossorigin: 'anonymous', rel: 'preload'
+
+    = javascript_pack_tag 'features/notifications', integrity: true, crossorigin: 'anonymous', rel: 'preload'
+    = javascript_pack_tag 'reducers/notifications', integrity: true, crossorigin: 'anonymous', rel: 'preload'
+
+    = javascript_pack_tag 'features/community_timeline', integrity: true, crossorigin: 'anonymous', rel: 'preload'
+
+    = javascript_pack_tag 'features/public_timeline', integrity: true, crossorigin: 'anonymous', rel: 'preload'
```

Comment:
These should have `as="script"` as well; it's explained what the browser does with this extra information [here](https://developers.google.com/web/updates/2016/03/link-rel-preload).